core/src/main/java/org/opensearch/remote/metadata/client/DataObjectRequest.java

```diff
@@ -39,10 +40,24 @@ protected DataObjectRequest(String index, String id, String tenantId) {
      * @param cmkRoleArn optional CMK role ARN (nullable)
      */
     protected DataObjectRequest(String index, String id, String tenantId, String cmkRoleArn) {
+        this(index, id, tenantId, cmkRoleArn, null);
```

Comment:
Please also update the 3-arg constructor (line 31) to pass an fifth (`null`) arg so it doesn't have to go through this one.